manager/controllers/app/fybrikapplication_controller_unit_test.go

```diff
@@ -1388,3 +1388,86 @@ func TestReadAndTransform(t *testing.T) {
+		Flow:         taxonomy.WriteFlow,
+		Requirements: app.DataRequirements{Interface: app.InterfaceDetails{Protocol: app.ArrowFlight}},
+	}
+	application.Spec.Data[2] = app.DataContext{
+		DataSetID:    "s3/new-dataset",
+		Flow:         taxonomy.ReadFlow,
+		Requirements: app.DataRequirements{Interface: app.InterfaceDetails{Protocol: app.ArrowFlight}},
+	}
+	application.SetGeneration(1)
+	application.SetUID("17")
+	// Objects to track in the fake client.
+	objs := []runtime.Object{
+		application,
+	}
+
+	// Register operator types with the runtime scheme.
+	s := utils.NewScheme(g)
+
+	// Create a fake client to mock API calls.
+	cl := fake.NewFakeClientWithScheme(s, objs...)
+
+	// Read module
+	readWriteModule := &app.FybrikModule{}
+	g.Expect(readObjectFromFile("../../testdata/unittests/module-read-write-parquet.yaml", readWriteModule)).NotTo(gomega.HaveOccurred())
+	readWriteModule.Namespace = utils.GetControllerNamespace()
+	g.Expect(cl.Create(context.TODO(), readWriteModule)).NotTo(gomega.HaveOccurred(), "the read module could not be created")
+
+	// Create a FybrikApplicationReconciler object with the scheme and fake client.
+	r := createTestFybrikApplicationController(cl, s)
+	g.Expect(r).NotTo(gomega.BeNil())
+
+	req := reconcile.Request{
+		NamespacedName: namespaced,
+	}
+
+	_, err := r.Reconcile(context.Background(), req)
+	g.Expect(err).To(gomega.BeNil())
+
+	err = cl.Get(context.TODO(), req.NamespacedName, application)
+	g.Expect(err).To(gomega.BeNil(), "Cannot fetch fybrikapplication")
+	g.Expect(getErrorMessages(application)).To(gomega.BeEmpty())
+	// check plotter creation
+	g.Expect(application.Status.Generated).ToNot(gomega.BeNil())
+	readOriginaDatalEndpoint := application.Status.AssetStates[application.Spec.Data[0].DataSetID].Endpoint
+	g.Expect(readOriginaDatalEndpoint).To(gomega.Not(gomega.BeNil()))
+	readNewDataEndpoint := application.Status.AssetStates[application.Spec.Data[2].DataSetID].Endpoint
+	g.Expect(readNewDataEndpoint).To(gomega.Not(gomega.BeNil()))
+	readOriginaDataConnectionMap := readNewDataEndpoint.AdditionalProperties.Items
+	g.Expect(readOriginaDataConnectionMap).To(gomega.HaveKey("fybrik-arrow-flight"))
+	readOriginaDataConfig := readOriginaDataConnectionMap["fybrik-arrow-flight"].(map[string]interface{})
+	g.Expect(readOriginaDataConfig["hostname"]).To(gomega.Equal("read-write-module"))
+	readNewDataConfig := readOriginaDataConnectionMap["fybrik-arrow-flight"].(map[string]interface{})
+	g.Expect(readNewDataConfig["hostname"]).To(gomega.Equal("read-write-module"))
+	plotterObjectKey := types.NamespacedName{
+		Namespace: application.Status.Generated.Namespace,
+		Name:      application.Status.Generated.Name,
+	}
+	plotter := &app.Plotter{}
+	err = cl.Get(context.Background(), plotterObjectKey, plotter)
+	g.Expect(err).NotTo(gomega.HaveOccurred())
+	g.Expect(plotter.Spec.Assets).To(gomega.HaveLen(2))
+	g.Expect(plotter.Spec.Templates).To(gomega.HaveLen(2)) // expect one template
```

Comment:
How many templates do you expect: 1 or 2?

Reply:
two, it is an old comment. I fixed that